policyengine_us/variables/gov/states/hi/tax/income/exemptions/hi_disabled_exemptions.py

```diff
@@ -24,20 +24,41 @@ def formula(tax_unit, period, parameters):
         aged_spouse = (
             tax_unit("age_spouse", period) >= p.aged_threshold
         ).astype(int)
-        disabled_head = tax_unit("head_is_disabled", period)
+        disabled_head = tax_unit("head_is_disabled", period).astype(int)
         disabled_exemption_disabled_head = disabled_head * p.disabled
         disabled_exemption_non_disabled_head = p.base * (1 + aged_head)
-        disabled_exemption_head = max_(
-            disabled_exemption_disabled_head,
+        disabled_exemption_head = where(
+            disabled_head,
+            max_(
+                disabled_exemption_disabled_head,
+                disabled_exemption_non_disabled_head,
+            ),
             disabled_exemption_non_disabled_head,
         )
         # Same for spouse.
-        disabled_spouse = tax_unit("spouse_is_disabled", period)
+        disabled_spouse = tax_unit("spouse_is_disabled", period).astype(int)
         disabled_exemption_disabled_spouse = disabled_spouse * p.disabled
         disabled_exemption_non_disabled_spouse = p.base * (1 + aged_spouse)
-        disabled_exemption_spouse = max_(
-            disabled_exemption_disabled_spouse,
+        disabled_exemption_spouse = where(
+            disabled_spouse,
+            max_(
+                disabled_exemption_disabled_spouse,
+                disabled_exemption_non_disabled_spouse,
+            ),
             disabled_exemption_non_disabled_spouse,
         )
 
-        return disabled_exemption_head + disabled_exemption_spouse
+        # if filling status is not joint, the disabled_exemption_spouse should be zero
+        # The taxpayer shall not take additional exemptions with regard to the taxpayer’s disability.
+        filing_status = tax_unit("filing_status", period)
+        disabled_exemption_spouse = where(
+            filing_status == filing_status.possible_values.JOINT,
+            disabled_exemption_spouse,
+            0,
+        )
+        no_disabled_head_or_spouse = (disabled_head + disabled_spouse) == 0
+        return where(
```

Comment:
why is this necessary? won't both parts of the sum be zero in this case?

Reply:
The exemption for a disabled taxpayer will be non-zero if the spouse is disabled and they are filing jointly. Thus, the formula's straightforward logic calculates the disabled exemption when disabled or the regular exemption if not. This line ensures a zero total exemption when both the head and spouse are not disabled, correcting the previous error.